stdlib/jvm/compile.mc

```diff
@@ -0,0 +1,57 @@
+include "mexpr/ast.mc"
+include "string.mc"
+include "jvm/ast.mc"
+
+lang MExprJVMCompile = MExprAst + JVMAst
+
+    -- go through AST and translate to JVM bytecode
+
+    sem toJSONExpr : JVMEnv -> Expr -> JVMEnv
+    sem toJSONExpr env =
+    | TmSeq {tms = tms} -> { env with bytecode = concat env.bytecode [(createBString "LDC" (_charSeq2String tms))]}
+    | TmConst { val = val } -> 
+        let bc = (match val with CInt { val = val } then (createBInt "LDC" val)
+        else never)
+        in { env with bytecode = concat env.bytecode [bc] }
+    | TmApp { lhs = lhs, rhs = rhs } -> 
+        match lhs with TmConst { val = val } then 
+            match val with CPrint _ then 
+                -- push this to stack
+                let this = { env with bytecode = concat env.bytecode [(createBApply "GETSTATIC" "java/lang/System" "out" "Ljava/io/PrintStream;")] } in
+                -- push args to stack
+                let args = (toJSONExpr this rhs) in
```

Comment:
You don't need parentheses around expressions in these cases - this also applies when they're an element in a sequence literal.